test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go

```diff
@@ -200,18 +194,46 @@ var _ = Describe("Mixedcpus", Ordered, func() {
 	})
 })
 
-func setup(ctx context.Context, profile *performancev2.PerformanceProfile) func(ctx context.Context) {
+func setup(ctx context.Context) func(ctx context.Context) {
+	var updateNeeded bool
+	profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+	Expect(err).ToNot(HaveOccurred())
 	initialProfile := profile.DeepCopy()
-	isolated, err := cpuset.Parse(string(*profile.Spec.CPU.Isolated))
+
+	if !profileutil.IsMixedCPUsEnabled(profile) {
+		isolated, err := cpuset.Parse(string(*profile.Spec.CPU.Isolated))
+		Expect(err).ToNot(HaveOccurred())
+
+		// arbitrary take the first one
+		sharedcpu := cpuset.New(isolated.List()[0])
+		testlog.Infof("shared cpu ids are: %q", sharedcpu.String())
+		updatedIsolated := isolated.Difference(sharedcpu)
+		profile.Spec.CPU.Isolated = cpuSetToPerformanceCPUSet(&updatedIsolated)
+		profile.Spec.CPU.Shared = cpuSetToPerformanceCPUSet(&sharedcpu)
+		profile.Spec.WorkloadHints.MixedCpus = pointer.Bool(true)
+		testlog.Infof("enable mixed cpus for profile %q", profile.Name)
+		updateNeeded = true
+	} else {
+		testlog.Infof("mixed cpus already enabled for profile %q", profile.Name)
+	}
+
+	workers, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
 	Expect(err).ToNot(HaveOccurred())
+	for _, worker := range workers {
+		if worker.Status.Capacity.Cpu().Size() == numberOfCoresThatRequiredCancelingSMTAlignment {
```

Comment:
maybe you want to have  a `>=` check rather than exact equality? exact equality is oftentimes fragile

Reply:
That was my thought process as well, but less than `numberOfCoresThatRequiredCancelingSMTAlignment` (4) the test would failed anyway.
But yes we can change that, no biggie